### PR TITLE
fix Huge Pages failing test

### LINKED ISSUE #121264: [Failing-Test] [Huge Pages] should remove resources for huge page sizes no longer supported
### Which jobs are failing?

The Huge pages test in both containerd/cri is consistently failing.  

### Which tests are failing?

- https://testgrid.k8s.io/sig-node-cri-o#ci-crio-cgroupv1-node-e2e-hugepages
- https://testgrid.k8s.io/sig-node-containerd#node-kubelet-containerd-hugepages

### Since when has it been failing?

Not sure.

### Testgrid link

_No response_

### Reason for failure (if possible)

```
{ failed [FAILED] capacity should contain resource hugepages-3Mi and huge pages with size 3Mi should be supported
Expected
    <v1.ResourceList | len:7>: {
        "cpu": {
            i: {value: 1, scale: 0},
            d: {Dec: nil},
            s: "1",
            Format: "DecimalSI",
        },
        "ephemeral-storage": {
            i: {value: 20869787648, scale: 0},
            d: {Dec: nil},
            s: "20380652Ki",
            Format: "BinarySI",
        },
        "hugepages-1Gi": {
            i: {value: 1073741824, scale: 0},
            d: {Dec: nil},
            s: "1Gi",
            Format: "BinarySI",
        },
        "hugepages-2Mi": {
            i: {value: 0, scale: 0},
            d: {Dec: nil},
            s: "0",
            Format: "DecimalSI",
        },
        "hugepages-3Mi": {
            i: {value: 9437184, scale: 0},
            d: {Dec: nil},
            s: "9Mi",
            Format: "BinarySI",
        },
        "memory": {
            i: {value: 7781175296, scale: 0},
            d: {Dec: nil},
            s: "7598804Ki",
            Format: "BinarySI",
        },
        "pods": {
            i: {value: 110, scale: 0},
            d: {Dec: nil},
            s: "110",
            Format: "DecimalSI",
        },
    }
to have {key: value}
    <map[interface {}]interface {} | len:1>: {
        <string>"hugepages-3Mi": <resource.Quantity>{
            i: {value: 9437184, scale: 0},
            d: {Dec: nil},
            s: "9Mi",
            Format: "BinarySI",
        },
    }
In [It] at: test/e2e_node/hugepages_test.go:217 @ 10/16/23 11:16:18.868
}
```

### Anything else we need to know?

There was a deprecation for ExpectEqual merged about a week ago on this same line.  

@carlory maybe you could take a look and see why this check is failing?

https://github.com/kubernetes/kubernetes/pull/121067 modified this line

### Relevant SIG(s)

/sig node

### PR DESCRIPTION
**Pull request by carlory**
#### What type of PR is this?

/kind failing-test

#### Which issue(s) this PR fixes:

Fixes #121264

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by kannon92 — COMMENTED** [review]
/hold
/lgtm

Waiting to confirm this fixes the test via presubmit (running now).

**Comment by kannon92** [comment]
Seeing "pull-crio-cgroupv1-node-e2e-hugepages" passing makes me confident that this PR can be merged.

/assign @SergeyKanzhelev

File: test/e2e_node/hugepages_test.go [hunk 1]
```diff
@@ -214,7 +214,11 @@ var _ = SIGDescribe("HugePages [Serial] [Feature:HugePages][NodeSpecialFeature:H
 		framework.ExpectNoError(err, "while getting node status")
 
 		ginkgo.By("Verifying that the node now supports huge pages with size 3Mi")
-		gomega.Expect(node.Status.Capacity).To(gomega.HaveKeyWithValue("hugepages-3Mi", resource.MustParse("9Mi")), "capacity should contain resource hugepages-3Mi and huge pages with size 3Mi should be supported")
+		value, ok := node.Status.Capacity["hugepages-3Mi"]
+		if !ok {
+			framework.Failf("capacity should contain resource hugepages-3Mi: %v", node.Status.Capacity)
+		}
+		gomega.Expect(value.String()).To(gomega.Equal("9Mi"), "huge pages with size 3Mi should be supported")
```
[the remark this answers is not in the document]
Comment (profnandaa) [reply test/e2e_node/hugepages_test.go:221]: Or actually that's the implementation and therefore this test case will always fail; should we then test for that failure and also redo the wording, re: `"huge pages with size 3Mi should be supported"`.

[the remark this answers is not in the document]
Comment (profnandaa) [reply test/e2e_node/hugepages_test.go:221]: Ok, got, this is historical... https://github.com/kubernetes/kubernetes/commit/2830827442358235a729b39e5eed8ec6cda29e12

but perhaps still needs a relook into it.

**Comment by kannon92** [comment]
Maybe have a fix for https://github.com/kubernetes/test-infra/pull/31051.  

It is not required for this to merge though.

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve

**Pull request merged**